src/nvim/ui.c

```diff
@@ -166,6 +166,115 @@ void ui_event(char *name, Array args)
   }
 }
 
+/// Retrieves attribute description from its id
+///
+/// @param attr_id attribute id
+Dictionary get_attr_by_id(Integer attr_id, Error *err)
```

Comment:
these new functions are in the UI module but don't have a `ui_` prefix. If they really belong there then they should have that prefix. Not sure if they do belong there though, `hl_` prefix might be better.

Reply:
should I move it to syntax.c ? or I can merge it with nvim_get_hl_by_id but I believe the current logic separation is clearer.